### DISCUSSION TITLE
Request from Next.js 13 (as front-end) to Hono server doesn't work

### DISCUSSION THREAD

**Discussion by GutoFrr** [honojs/hono#1370] [Q&A]
In my Hono server with Node.js I have this controller to a get method. The path to it is 'http://localhost:5000/api/products'

```tsx
async function getProducts(c: Context) {
  const products = await prisma.product.findMany({
    select: {
      ...PRODUCT_FIELDS,
      user: {
        select: {
          name: true,
        },
      },
    },
  });
  if (products.length === 0) {
    return c.json({ message: 'Não há produtos cadastrados...' }, 404);
  }

  return c.json({ products }, 200);
}
```

The Next.js data fetching function is this one:

```tsx
async function getProducts() {
  try {
    const res = await fetch('http://localhost:5000/api/products', {
      cache: 'no-store',
    });

    const data = await res.json();
    console.log(data);

    return data.products as IProduct[];
  } catch (error) {
    console.log(error);
  }
}
```

The error is this:
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED ::1:5000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 5000
  }
}

In Next.js, I can fetch the data successfully as a client function (useState and useEffect), but I can't with a server-side request, as it is described above. The request route doesn't require authentication, any user should be able to fetch the data.

Is important to remind that my Next.js front-end is not connected with Hono RPC. The server (Hono) and the client (Next.js) are in separate folders.

Well, my question here is how to connect an outside client to a Hono server. Thanks for helping!

**Comment by Kallarari** [comment]
this is an common problem in hono!

**Reply by GutoFrr** [reply]
I hope this can be fixed...

**Reply by Kallarari** [reply]
sad

**Comment by GutoFrr** [comment]
I fixed it... It was a silly issue. I did not know I needed to define the hostname in the serve function.
